Docs: Slight clarification of what is durable in sharding. (#27582)
akka/akka: akka-docs/src/main/paradox/cluster-sharding.md

akka.cluster.sharding.state-store-mode = ddata
```

The state of the `ShardCoordinator` is replicated across the cluster but is not durable, not stored to disk.
The `ShardCoordinator` state replication is handled by @ref:[Distributed Data](distributed-data.md) with `WriteMajority`/`ReadMajority` consistency.
When all nodes in the cluster have been stopped, the state is no longer needed and dropped.

The state of [Remembering Entities](#cluster-sharding-remembering) is durable and stored to

> disk. Thus stored entities are restarted after a complete cluster restart.

Reviewer: Perhaps in this Kubernetes world...
```suggestion
disk. This means remembered entities are restarted even after a complete (non-rolling) cluster restart when the disk is still available.
```

Cluster Sharding is using its own Distributed Data `Replicator` per node. 

> If using roles with sharding, the `Replicator` is per node role which enables a subset of

Reviewer: ```suggestion
If using roles with sharding there is one `Replicator` per role, which enables a subset of
```
